UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -621,6 +621,20 @@ public virtual void ServerPerformInteraction(AimApply interaction)
 			{
 				FiringPin.ServerBehaviour(interaction, isSuicide);
 			}
+
+			if (interaction.Intent == Intent.Harm && interaction.UsedObject == gameObject)
+			{
+				List<ItemSlot> hands = interaction.PerformerPlayerScript.DynamicItemStorage.GetHandSlots();
+				hands.Remove(interaction.PerformerPlayerScript.DynamicItemStorage.GetActiveHandSlot());
```

Comment:
this removes the active handslot from the newly created list of all handslots, so the foreach loop doesnt iterate over the gun thats triggering the dual wield interaction because that would trigger the dual wielding code twice
so it shoudnt really effect anything?

Reply:
ok cool